### PR TITLE
Allocation QoS [5091]

### PR DESCRIPTION
**Pull request by MiguelCompany**

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/57/)

File: docs/xmlprofiles.rst [hunk 1]
```diff
@@ -527,6 +534,51 @@ List with the possible configuration parameter:
...
+       participant will connect to.
+     - ``UInt32``
+     - 4
+   * - ``<max_multicast_locators>``
+     - Maximum number of multicast locators expected on a remote entity.
+       May be set to zero to disable multicast traffic.
+     - ``UInt32``
+     - 1
+   * - ``<total_participants>``
+     - Publisher :ref:`CommonAlloc` related to the total number of participants in the domain (local and remote).
```
Comment (richiware) [inline docs/xmlprofiles.rst]:
```suggestion
     - Participant :ref:`CommonAlloc` related to the total number of participants in the domain (local and remote).
```

File: docs/xmlprofiles.rst [hunk 2]
```diff
@@ -527,6 +534,51 @@ List with the possible configuration parameter:
...
+     - :ref:`CommonAlloc`
+     -
+   * - ``<total_readers>``
+     - Publisher :ref:`CommonAlloc` related to the total number of readers in the domain (local and remote).
```
Comment (richiware) [inline docs/xmlprofiles.rst]: Maybe for users will be easier to set the maximum readers per participant.

```suggestion
     - Participant :ref:`CommonAlloc` related to the total number of readers in the domain (local and remote).
```

File: docs/xmlprofiles.rst [hunk 3]
```diff
@@ -527,6 +534,51 @@ List with the possible configuration parameter:
...
+     - :ref:`CommonAlloc`
+     -
+   * - ``<total_writers>``
+     - Publisher :ref:`CommonAlloc` related to the total number of writers in the domain (local and remote).
```
Comment (richiware) [inline docs/xmlprofiles.rst]: Maybe for users will be easier to set the maximum writers per participant.
```suggestion
     - Participant :ref:`CommonAlloc` related to the total number of writers in the domain (local and remote).
```

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/66/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/72/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/73/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/74/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/75/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/76/)

File: docs/xmlprofiles.rst [hunk 4]
```diff
@@ -1274,6 +1285,39 @@ Throughput Configuration allows to limit the output bandwidth.
...
+
+Allocation Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Allocation Configuration allows to control the allocation behavior of internal collections for which the number
+of elements depends on the number of entities in the system.
+
+For instance, there are collections inside a publisher which depend on the number of subscribers matching with it.
+
+.. TODO: Add reference to allocation tuning section
```
Comment (raquelalvarezbanos) [inline docs/xmlprofiles.rst]: Should we address this "TODO" comment as well as part of this PR?

File: docs/xmlprofiles.rst [hunk 5]
```diff
@@ -660,6 +667,39 @@ List with the possible configuration parameter:
...
+   * - Name
+     - Description
+     - Values
+     - Default
+   * - ``<max_unicast_locators>``
+     - Maximum number of unicast locators expected on a remote entity.
+       It is recommended to use the maximum number of network interfaces found on any machine the
+       participant will connect to.
+     - ``UInt32``
+     - 4
```
Comment (raquelalvarezbanos) [inline docs/xmlprofiles.rst]: Just out of interest, why is the default value 4?

File: Makefile [hunk 6]
```diff
@@ -78,8 +78,8 @@ test:
 	grep -Rn "code-block::.*xml" docs; test "$$?" -ne "0"
 	@echo
 	@echo "Check xml code-block finished."
-	mkdir -p build/code
-	cd build/code && cmake -G Ninja ../../code
```
Comment (raquelalvarezbanos) [inline Makefile]: Why was this removed?

File: docs/xmlprofiles.rst (hunk 4 above)
Comment (MiguelCompany) [reply docs/xmlprofiles.rst]: The plan is to do it as part of the release PR.

File: docs/xmlprofiles.rst (hunk 5 above)
Comment (MiguelCompany) [reply docs/xmlprofiles.rst]: We thought it is a good compromise value. It is not usual to have more than 4 network interfaces.

File: Makefile (hunk 6 above)
Comment (MiguelCompany) [reply Makefile]: Ninja is not installed in the docker image anymore

**Comment by richiware** [comment]
Build status: * Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Docs/4/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/13/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/18/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/31/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/34/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/35/)
